ci/integration.js

```diff
@@ -0,0 +1,236 @@
+const child_process = require("child_process");
+const { randomBytes } = require("crypto");
+const path = require("path");
+const fs = require("fs");
+
+global.XMLHttpRequest = require("xmlhttprequest").XMLHttpRequest;
+
+const { ajax } = require("rxjs/ajax");
+
+const rxJupyter = require("rx-jupyter");
+
+const _ = require("lodash");
+
+console.log("running bookstore integration tests");
+
+async function genToken(byteLength = 32) {
+  return new Promise((resolve, reject) => {
+    randomBytes(byteLength, (err, buffer) => {
+      if (err) {
+        reject(err);
+        return;
+      }
+
+      resolve(buffer.toString("hex"));
+    });
+  });
+}
+
+const sleep = timeout =>
+  new Promise((resolve, reject) => setTimeout(resolve, timeout));
+
+// Catch all rogue promise rejections to fail CI
+process.on("unhandledRejection", error => {
+  console.log("unhandledRejection", error.message);
+  process.exit(2);
+});
+
+var Minio = require("minio");
+
+const main = async () => {
+  const jupyterToken = await genToken();
+  const jupyterPort = 9988;
+  const jupyterEndpoint = `http://127.0.0.1:${jupyterPort}`;
+
+  const bucketName = "bookstore";
+  // Optional according to minio docs, likely here for AWS compat
+  const regionName = "us-east-1";
+
+  const s3Config = {
+    endPoint: "127.0.0.1",
+    port: 9000,
+    useSSL: false,
+    accessKey: "ONLY_ON_CIRCLE",
+    secretKey: "CAN_WE_DO_THIS"
+  };
+
+  // Instantiate the minio client with the endpoint
+  // and access keys as shown below.
+  var minioClient = new Minio.Client(s3Config);
+
+  const madeBucket = await new Promise((resolve, reject) => {
+    minioClient.makeBucket(bucketName, regionName, err => {
+      if (err) {
+        reject(err);
+        return;
+      }
+      resolve();
+    });
+  });
+  console.log(`Created bucket ${bucketName}`);
+
+  const jupyter = child_process.spawn(
+    "jupyter",
+    [
+      "notebook",
+      "--no-browser",
+      `--NotebookApp.token=${jupyterToken}`,
+      `--NotebookApp.disable_check_xsrf=True`,
+      `--port=${jupyterPort}`,
+      `--ip=127.0.0.1`
+    ],
+    { cwd: __dirname }
+  );
+
+  ////// Refactor me later, streams are a bit messy with async await
+  // Check to see that jupyter is up
+  let jupyterUp = false;
+
```

Comment:
I could make this all a bit cleaner with rxjs + spawn-rx. As long as folks are cool with it for now, I'm ok leaving all this since it's an integration test.

Reply:
Let's mention rxjs + spawn-rx in the refactor comments. Just in case another contributor does the refactor

Reply:
Ok, will do.